Add extra content headers from Croquemort

AntoineAugusti opened the pull request:
Some content headers would be useful to have to update resources' metadata on udata.

Content-Type and Content-Length could be used to set the MIME type and the filesize of resources, enabling a larger usage of CSVAPI for remote resources.

AntoineAugusti commented on udata_croquemort/tests.py:
```diff
@@ -90,11 +92,14 @@ def test_returned_metadata(self, httpretty):
         url = self.resource.url
         metadata = {
-            'content-type': 'text/html; charset=utf-8',
+            'content-type': 'text/html',
```
Charset is removed from Content-Type on Croquemort https://github.com/etalab/croquemort/pull/202

AntoineAugusti commented on udata_croquemort/checker.py:
```diff
@@ -24,13 +24,21 @@ def _format_response(self, response):
             }
 
+            for header in [
+                'content-type', 'content-length', 'content-md5', 'charset'
```
We could pull more headers https://github.com/etalab/croquemort/blob/eb972ebee6a17c16ac838edffdaa1d3c3ea23b6e/croquemort/storages.py#L12-L16

AntoineAugusti commented on udata_croquemort/checker.py:
```diff
@@ -24,13 +24,21 @@ def _format_response(self, response):
+            for header in [
+                'content-type', 'content-length', 'content-md5', 'charset'
+            ]:
+                result[f"metadata:{header}"] = response.get(header)
```
Not sure of the prefix

abulte replied:
Maybe `check-metadata:` since we already use `check:` for croquemort related stuff.

abulte replied:
or `check-headers`

abulte commented:
Thanks @antoineaugusti!

@geoffreyaldebert should we invest in this or instead rely on your upcoming implementation?

AntoineAugusti commented:
I changed the prefix and added a note in the CHANGELOG.

I'm waiting on your green light to do the work on udata, around these lines https://github.com/opendatateam/udata/blob/25fa4ee05726e101539ed2d152b14ac842fdb6e7/udata/linkchecker/checker.py#L69-L70

geoffreyaldebert commented:
Definetely a quickwin. Agree with you all. So I'm OK.

 @AntoineAugusti can you add "Content-Disposition" header that is referring the filename (and its extension) and could be very useful some times (especially when content-type is defined to application/octet-stream)

abulte commented on udata_croquemort/checker.py:
```diff
@@ -24,13 +24,21 @@ def _format_response(self, response):
+            for header in [
+                'content-type', 'content-length', 'content-md5', 'charset'
+            ]:
+                result[f"check-headers:{header}"] = response.get(header)
```
After reading https://github.com/opendatateam/udata/blob/25fa4ee05726e101539ed2d152b14ac842fdb6e7/udata/linkchecker/checker.py#L8, maybe we could settle on `check:headers:xxx`. This would allow this PR to work w/o modification on udata's side. WDYT?

AntoineAugusti commented:
I added the `Content-Disposition` header as requested @geoffreyaldebert. As discussed on Slack, we chose the `check:headers:` prefix for these new headers in order to avoid changing code on `udata`.

I'll open a PR on tabular preview to use these new extra headers if they are present to enable CSVAPI for these resources

abulte commented on udata_croquemort/checker.py:
```diff
@@ -24,13 +24,22 @@ def _format_response(self, response):
+                'content-type', 'content-length', 'content-md5', 'charset',
+                'content-disposition'
+            ]:
+                result[f"check:headers:{header}"] = response.get(header)
```
maybe add a safe guard here? `if in` or `.get`

abulte replied:
🙈

AntoineAugusti replied:
`.get` already has a safeguard, it's not a direct key access with `dict[key]`
```
Python 3.7.7 (default, Mar 10 2020, 15:43:33) 
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> print({}.get('a'))
None
```

abulte commented on CHANGELOG.md:
```diff
@@ -2,7 +2,7 @@
 ## Current (in progress)
 
-- Nothing yet
+- Expose new HTTP headers related to content (`content-type`, `content-length`, `charset`, `content-md5`, `content-disposition`)
```
we usually add a link to the PR
